### PR TITLE
More useful debugging output in securedrop-admin

### LINKED ISSUE #2563: Verbose/debug mode in securedrop-admin should show CalledProcessError output
## Description

We have a few cases where we catch `subprocess.CalledProcessError` in `securedrop-admin` and print a generic error message. We should enable an admin that runs `securedrop-admin` in verbose mode to see the full output for debugging purposes.

## User Stories

As a SecureDrop engineer, I want detailed tracebacks when errors occur so that I know what the underlying issues are in the code. 

As a SecureDrop administrator, I want to be able to send detailed logs to the SecureDrop team so they can figure out what is going on.

### PR DESCRIPTION
**Pull request by redshiftzero**
## Status

Ready for review

## Description of Changes

Fixes #2563

Changes proposed in this pull request:
 - more detailed messages printed to screen when verbose mode set for `subprocess.CalledProcessError`

## Testing

Easiest way to test in your Tails VM is to turn off networking and you should see a nice detailed error message. 

## Deployment

Admins will check out in their workstations in the next release

## Checklist

Tails only, needs to be manually tested

### CODE REVIEW

**Comment by dachary** [comment]
@redshiftzero this review is in commits instead of words ;-) I figured it might be possible to test securedrop-admin, found a few problems on the way and fixed them to verify there were no pitfalls creating problems with testing. It's more comfortable than manual testing. Feel free to rebase / squash as you like: these commits should be treated as comments rather than definitive recommendations.

**Comment by dachary** [comment]
rebased & repushed

**Comment by dachary** [comment]
@redshiftzero the unexpected number of commits fixing various test related problems reflects how difficult it was to get a stable run. I believe the outcome is good and leads to more stable tests, both in circleci and locally.

File: .circleci/config.yml [hunk 1]
```diff
@@ -160,9 +160,8 @@ jobs:
       - run: mkdir -p ~/test-results
 
       - run: |
-          set +o pipefail
           export TESTFILES=$(cd securedrop; circleci tests glob 'tests/test*py' 'tests/**/test*py' |circleci tests split --split-by=timings |xargs echo)
-          docker rm -f securedrop-test |true
+          docker rm -f securedrop-test || true
```
Comment (conorsch) [inline .circleci/config.yml]: :ok_hand:

**Comment by conorsch** [comment]
Dang, @dachary, really nice testing additions. Since this was originally @redshiftzero's PR, @redshiftzero can't formally approve, and if you did it would mean little review-wise at this point. :grin:

Will take this for a spin in Tails this week and try to get it in.

**Comment by dachary** [comment]
I'll repush again once to verify it actually is stable.

**Comment by dachary** [comment]
> Will take this for a spin in Tails this week and try to get it in.

Thanks ! :-D

**Comment by conorsch** [comment]
Looks like the fix on the Circle CI config now shows conflicts, due to merge of #2588. Will rebase all changes to resolve.

**Comment by conorsch** [comment]
Hmm, the conversion of `securedrop-admin` to a symlink broke the 9p mounting of the repo in my Tails VM:

```
amnesia@amnesia:~/Persistent/securedrop$ cat securedrop-admin
cat: securedrop-admin: Too many levels of symbolic links
```
Is that move required for the tests to run, @dachary ? The commit message doesn't have any details.

**Comment by dachary** [comment]
> Is that move required for the tests to run ?

If ``securedrop-admin`` is not in the securedrop directory, the tests are no longer contained in this directory. It means that when we run pytest from the securedrop directory and it has to fetch the script from the parent, which is not impossible or blocking but very unusual. There is another side effect which is that ``securedrop-admin`` is not included in /var/www/securedrop and when the tests run in staging we would have to explicitly exclude the ``securedrop-admin`` tests. And yet another side effect which is that the Dockerfile that runs make test does not include ``securedrop-admin`` and will also fail to run the tests.

Another approach could be to isolate securedrop-admin and treat it as a completely separated script. Create a test directory for it, targets to run pytests for it, requirements and all that. This is a little heavy but not absurd: securedrop-admin really is **not** part of the app, it is a deployment helper.

**Comment by dachary** [comment]
Maybe the right thing to do would simply be to move securedrop-admin into securedrop and update the docs so people use securedrop/securedrop-admin instead of securedrop-admin. And not have this annoying symlink.

**Comment by dachary** [comment]
Looks like the machine hosting the travis test gave up https://travis-ci.org/freedomofpress/securedrop/builds/319806654?utm_source=github_status&utm_medium=notification

<pre>
"Setting up libtie-ixhash-perl (1.23-1) ...", "Setting up hardening-includes (2.5ubuntu2.
The command "ansible-playbook -i inventory -vv --connection=local install_files/ansible-base/securedrop-development.yml
  " failed and exited with 1 during .
</pre>

**Comment by dachary** [comment]
There is a much simpler solution. I'll remove the symlink and replace it with a script:

<pre>
#!/bin/bash
exec securedrop/securedrop-admin "$@"
</pre>

which will solve the symlink situation and keep the script inside the securedrop directory, together will all other python scripts. It does not really belong there in the sense that it's not part of the app. It only belongs because it is a python script and all the infrastructure for distribution and testing is in this place. Do you think this is acceptable @conorsch ?

**Comment by conorsch** [comment]
*smacks forehead* That's a fine compromise, @dachary! Preferably with a comment in the wrapper script to explain its necessity. 

For the record, I do think it's worthwhile to port the tests to a more standard location, but that work should not block merge here—we have plenty of testing work to do in the coming months!

**Comment by dachary** [comment]
transient error https://travis-ci.org/freedomofpress/securedrop/builds/320435888?utm_source=github_status&utm_medium=notification

<pre>
...", "Unpacking python3-chardet (2.2.1-2~ubuntu1) ...", "Selecting previously unselected package python3-six.", "Preparing to unpack .../python3-six_1.5.2-1ubuntu1_all.deb ...", "Unpacking python3-six (1.5.2-1ubuntu1) ...", "Selecting previously unselected package python3-debian.", "Prep
The command "ansible-playbook -i inventory -vv --connection=local install_files/ansible-base/securedrop-development.yml
  " failed and exited with 1 during .
Your build has been stopped.
</pre>

**Comment by dachary** [comment]
Rebased to address the .circleci conflict

**Pull request merged**